common/pldm/src/protocol/version.rs

```diff
@@ -0,0 +1,265 @@
+// Licensed under the Apache-2.0 license
+
+use core::{convert::TryFrom, fmt};
+
+pub type Ver32 = u32;
+pub type VersionCheckSum = u32;
+pub type ProtocolVersionStr = &'static str;
+
+// The PLDM base protocol version 1.1.0
+pub const PLDM_BASE_PROTOCOL_VERSION: ProtocolVersionStr = "1.1.0";
+
+// PLDM firmware update protocol 1.3.0
+pub const PLDM_FW_UPDATE_PROTOCOL_VERSION: ProtocolVersionStr = "1.3.0";
+
+/// PLDM version structure. Ver32 encoding
+///
+/// The "major," "minor," and "update" bytes are BCD-encoded, and each byte holds two BCD
+/// digits. The "alpha" byte holds an optional alphanumeric character extension that is encoded using the
+/// ISO/IEC 8859-1 Character Set. The value 0xF in the most-significant nibble of a BCD-encoded value indicates that the most
+/// significant nibble should be ignored and the overall field treated as a single-digit value. Software
+/// or utilities that display the number should display only a single digit and should not put in a
+/// leading "0" when displaying the number.
+///
+/// A value of 0xFF in the "update" field indicates that the entire field is not present. 0xFF is not
+/// allowed as a value for the "major" or "minor" fields. Software or utilities that display the version
+/// number should not display any characters for this field.
+///
+/// For example:
+/// - Version 3.7.10a → 0xF3F71061
+///
+/// - Version 3.1 → 0xF3F1FF00
+/// - Version 1.0a → 0xF1F0FF61
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[repr(C)]
+pub struct PldmVersion {
+    pub alpha: u8,
+    pub update: u8,
+    pub minor: u8,
+    pub major: u8,
+}
+
+// Convert from version string to PldmVersion
+impl TryFrom<ProtocolVersionStr> for PldmVersion {
+    type Error = String;
+
+    fn try_from(version: ProtocolVersionStr) -> Result<Self, Self::Error> {
+        // alpha can be attached to either minor or update
+
+        let mut version_parts = version.split('.').collect::<Vec<ProtocolVersionStr>>();
+        if version_parts.len() < 2 || version_parts.len() > 3 {
+            return Err("Invalid version format".to_string());
+        }
+
+        // Extract alpha, update, minor and major from the version string
+        let alpha = if version_parts[1].chars().last().unwrap().is_alphabetic() {
```

Comment:
I think it would be better to check that it is not empty before we call `unwrap()` to avoid the possible panic.